content/en/blog/_posts/2021-07-19-alpha-swap-support.md

```diff
@@ -0,0 +1,139 @@
+# Node alpha swap memory support
+
+---
+layout: blog
+title: 'Alpha swap support'
+date: 2021-07-19
+slug: alpha-swap-support
+---
+
+**Author:** Elana Hashman (Red Hat)
+
+The 1.22 release introduces alpha support for configuring swap memory usage for
+Kubernetes workloads on a per-node basis.
+
+In prior releases, Kubernetes did not support the use of swap memory on Linux,
+as it is difficult to provide guarantees and account for pod memory utilization
+when swap is involved. As part of Kubernetes' earlier design, swap support was
+considered out of scope, and a kubelet would by default fail to start if swap
+was detected on a node.
+
+However, there are a number of [use cases](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#user-stories)
+that would benefit from Kubernetes nodes supporting swap, including  improved
+node stability, better support for applications with high memory overhead but
+smaller working sets, the use of memory-constrained devices, and memory
+flexibility.
+
+Hence, over the past two release, SIG Node has been working to gather
+appropriate use cases and feedback, and propose a design for adding swap
+support to nodes in a controlled, predictable manner so that Kubernetes users
+can perform testing and provide data to continue building cluster capabilities
+on top of swap. The alpha graduation of swap memory support for nodes is our
+first milestone towards this goal!
+
+## How does it work?
+
+There are a number of possible ways that one could envision swap use on a node.
+To keep the scope manageable for this initial implementation, when swap is
+already provisioned and available on a node, [we have proposed](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#proposal)
+the kubelet should be able to be configured such that:
+
+- It can start with swap on.
+- It will direct the CRI to allocate Kubernetes workloads zero swap memory by
+  default.
+- It will have configuration options to specify swap utilization for the entire
+  node.
+
+Swap configuration on a node is exposed to a cluster admin via the
+[`memorySwap` in the KubeletConfiguration](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#kubeconfig-addition).
+The admin can specify the node's behaviour in the presence of swap memory by
+setting `memorySwap.swapBehavior`.
+
+This is possible through [the addition of a `memory_swap_limit_in_bytes`
+field](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#cri-changes)
```

Comment:
I wouldn't hyperlink here - to me I think it'll confuse some readers enough to be unhelpful.